cogs/hideout/boost_roles.py

```diff
@@ -0,0 +1,265 @@
+import logging
+from io import BytesIO
+from typing import Optional, Union
+
+import discord
+from discord import app_commands
+from discord.utils import MISSING
+from emoji import is_emoji
+
+from utils import HideoutCog
+
+from . import HideoutManager
+
+Interaction = discord.Interaction[HideoutManager]
+log = logging.getLogger(__name__)
+
+
+class BoostRoles(HideoutCog):
+    @staticmethod
+    async def get_emoji(input: str) -> Optional[Union[str, bytes]]:
+        if is_emoji(input):
+            return input
+
+        partial = discord.PartialEmoji.from_str(input)
+        if partial.is_unicode_emoji():
+            return input
+
+        elif partial.id is None:
+            return None
+
+        return await partial.read()
```

Comment:
Due to the nature of the `is_unicode_emoji` check - as seen [here](https://github.com/Rapptz/discord.py/blob/v2.3.x/discord/partial_emoji.py#L213-L215), this function is best to be rewritten as following. If not checked this way, this could allow any arbitrary string to be passed due to the nature of the aforementioned method

```suggestion
    async def get_emoji(input: str) -> Optional[Union[str, bytes]]:
        partial = discord.PartialEmoji.from_str(input)
        
        if partial.is_unicode_emoji():
            if is_emoji(partial.name):
                return partial.name

        else:
            return await partial.read()
```